File: 1_javabase/src/main/java/filetool/MyPropertiesUtil.java
package filetool;

import java.io.FileReader;
import java.io.IOException;
import java.util.Properties;

/*
    读取配置文件工具类
 */
public class MyPropertiesUtil {
    public static String load(String propertiesName, String key) throws IOException {
        Properties properties = new Properties();
        properties.load(new FileReader(propertiesName));
        String strValue = properties.getProperty(key);
        return strValue;
    }

    public static void main(String[] args) throws IOException {
        String str = MyPropertiesUtil.load("D:\\workhouse\\Modules\\javabase\\src\\main\\resources\\condition.properties", "condition.params.json");
        System.out.println(str);
    }
}

File: spark实时项目二(待完善)/src/main/resources/gmall2020_realtime/src/main/scala/xuexi/utils/MyPropertiesUtil.scala
//package 项目二实时项目.gmall2020-realtime.src.main.scala.xuexi.utils

import java.io.InputStreamReader
import java.nio.charset.StandardCharsets
import java.util.Properties

/**
  * Author: lizhong.liu
  * Date: 2020/9/12
  * Desc: 读取配置文件的工具类
  */
object MyPropertiesUtil {
  def load(propertiesName: String): Properties = {
    val prop: Properties = new Properties()
    prop.load(new InputStreamReader(
      Thread.currentThread().getContextClassLoader.getResourceAsStream(propertiesName), StandardCharsets.UTF_8))
    prop
  }

  def main(args: Array[String]): Unit = {
    val prop: Properties = MyPropertiesUtil.load("config.properties")
    val str: String = prop.getProperty("kafka.broker.list")
    println(str)
  }
}
